frontend/src/components/SearchResults.js

```diff
@@ -35,8 +37,47 @@ class SearchResults extends Component {
     return yearArr;
   });
 
+  setGradeFilters = filterEvent => {
+    let grade = parseInt(filterEvent.target.parentElement.innerText);
+    let gradeFilterArr = this.state.gradeFilters;
+    let idx = gradeFilterArr.indexOf(grade);
+    if (idx == -1) {
+      gradeFilterArr.push(grade);
+      this.setState({ gradeFilters: gradeFilterArr });
```

Comment:
Move the `setState` call to after the `else` case. If you look at the else case, you're not actually removing the grade in `state`, you're only removing it in your local variable.